credentials/generate-revocation-set.py

```diff
@@ -377,40 +450,64 @@ def main(use_main_net_dcld: str, use_test_net_dcld: str, use_main_net_http: bool
+        # TODO: 8. Validate CRL as per Section 6.3 of RFC 5280
+
+        # 9. decide on certificate authority name and AKID
+        if revocation_point["isPAA"] and not is_self_signed_certificate(crl_signer_certificate):
+            certificate_authority_name_b64 = get_subject_b64(paa_certificate_object)
+            certificate_akid = get_skid(paa_certificate_object)
+        elif crl_signer_delegator_cert:
+            certificate_authority_name_b64 = get_subject_b64(crl_signer_delegator_cert)
+            certificate_akid = get_skid(crl_signer_delegator_cert)
+        else:
+            certificate_authority_name_b64 = get_subject_b64(crl_signer_certificate)
+            certificate_akid = get_skid(crl_signer_certificate)
+
+        # validate issuer skid matchces with the one in revocation points
+        certificate_akid_hex = ''.join('{:02X}'.format(x) for x in certificate_akid)
+
+        logging.debug(f"Certificate Authority Name: {certificate_authority_name_b64}")
+        logging.debug(f"Certificate AKID: {certificate_akid_hex}")
+        logging.debug(f"revocation_point['issuerSubjectKeyID']: {revocation_point['issuerSubjectKeyID']}")
+
+        if revocation_point["issuerSubjectKeyID"] != certificate_akid_hex:
+            logging.warning("CRL Issuer Subject Key ID is not CRL Signer Subject Key ID, continue...")
+            continue
 
         serialnumber_list = []
         # 10. Iterate through the Revoked Certificates List
         for revoked_cert in crl_file:
-            # a.
             try:
                 revoked_cert_issuer = revoked_cert.extensions.get_extension_for_oid(
                     x509.CRLEntryExtensionOID.CERTIFICATE_ISSUER).value.get_values_for_type(x509.DirectoryName).value
 
                 if revoked_cert_issuer is not None:
-                    if revoked_cert_issuer != certificate_authority_name:
+                    # check if this really are the same thing
+                    if revoked_cert_issuer != certificate_authority_name_b64:
                         logging.warning("CRL Issuer is not CRL File Issuer, continue...")
                         continue
             except Exception:
                 pass
 
-            # b.
-            # TODO: Verify that the certificate chain of the entry is linking to the same PAA
-            #       that issued the CRLSignerCertificate for this entry, including path through
-            #       CRLSignerDelegator if present. If the PAAs under which were issued the certificate
-            #       and the CRLSignerCertificate are different, ignore the entry.
-
-            # c. and d.
             serialnumber_list.append(bytes(str('{:02X}'.format(revoked_cert.serial_number)), 'utf-8').decode('utf-8'))
 
-        issuer_name = base64.b64encode(crl_file.issuer.public_bytes()).decode('utf-8')
+        # No point in creating an entry which has no revoked serial numbers
+        if len(serialnumber_list) == 0:
+            continue
```

Comment:
Let's generate 1 set for each, even if empty.

Reply:
It was generating before, I though its an overhead so added that check, removed.